### DISCUSSION TITLE
NO way to change font for all

### DISCUSSION THREAD

**Discussion by dejudicibus** [TomSchimansky/CustomTkinter#2156] [General]
It looks like there is no way to change the font family for the whole window and all its widgets. I tried

```
class App(ctk.CTk):
    def __init__(self):
        super().__init__()
        self.defaultFont = tk.font.nametofont("TkDefaultFont")
        self.defaultFont.configure(family="Roboto", size=12)
        self.configure(font=self.defaultFont)
```
but it does not work.

**Comment by ElectricCandlelight** [comment]
https://customtkinter.tomschimansky.com/documentation/utility-classes/font

**Comment by dejudicibus** [comment]
I already know that page. It does not answer my question.

Dario de Judicibus

**Reply by ElectricCandlelight** [reply]
If you follow the code example on that page it will work. The code you posted isn't using CTkFont

**Reply by dejudicibus** [reply]
I tried and it does NOT work.

```
        self.defaultFont = ctk.CTkFont(family="Roboto", size=12)
        self.configure(font=self.defaultFont)
```
It looks like there is no way to change the default font for the whole window. 
The example shows how to change the font for a SINGLE widget. 
I need to change the font of ALL widgets at once.

**Comment by dejudicibus** [comment]
Ok I’ll try and let you know.

Dario de Judicibus
Scrittore, Roma (Italia)

**Comment by ElectricCandlelight** [comment]
```Python

import customtkinter

customtkinter.ThemeManager.theme["CTkFont"]["family"] = "Comic Sans MS"
customtkinter.ThemeManager.theme["CTkFont"]["size"] = 20

class App(customtkinter.CTk):
    def __init__(self):
        super().__init__()

        self.label = customtkinter.CTkLabel(master=self, text="Hello World")
        self.label.pack()

        self.button = customtkinter.CTkButton(master=self, text="Button")
        self.button.pack()

app = App()
app.geometry("400x240")
app.mainloop()

```

**Comment by dejudicibus** [comment]
Thank you. That works.
--DdJ

On Fri, Dec 22, 2023 at 2:50 AM ElectricCandlelight <
***@***.***> wrote:

> import customtkinter
> customtkinter.ThemeManager.theme["CTkFont"]["family"] = "Comic Sans MS"customtkinter.ThemeManager.theme["CTkFont"]["size"] = 20
> class App(customtkinter.CTk):
>     def __init__(self):
>         super().__init__()
>
>         self.label = customtkinter.CTkLabel(master=self, text="Hello World")
>         self.label.pack()
>
>         self.button = customtkinter.CTkButton(master=self, text="Button")
>         self.button.pack()
> app = App()app.geometry("400x240")app.mainloop()
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TomSchimansky/CustomTkinter/discussions/2156#discussioncomment-7923637>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAK5PECEAKUE2LPZ3AM4RALYKTRP5AVCNFSM6AAAAABA4P7MXWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSMRTGYZTO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> .com>
>


-- 
Dario de Judicibus, Rome, Italy (EU)
Site: https://www.dejudicibus.it, https://genealogia.dejudicibus.it
Blog: https://www.lindipendente.eu
Book: https://www.lalamanera.it, https://www.lasorgentedeimondi.it
